tests/sparseml/transformers/test_transformers.py

```diff
@@ -0,0 +1,198 @@
+        )
+    output = ort_sess.run(
+        None,
+        {"input_ids": input_0, "attention_mask": input_1, "token_type_ids": input_2},
+    )
+    return output
+
+
+def _compare_onnx_models(model1, model2):
+    optional_nodes_model1 = [
+        "If",
+        "Equal",
+        "Gather",
+        "Shape",
+        # ops above are those which are used in the
+        # original graph to create logits and softmax heads
+        "Constant",
+        "Cast",
+    ]  # ops above are the remaining optional nodes
+    optional_nodes_model2 = [
+        "Constant",
+        "Squeeze",
+    ]  # ops above are
+    # used in the original graph to create
+    # logits and softmax heads
+
+    nodes1 = model1.graph.node
+    nodes1_names = [node.name for node in nodes1]
+
+    nodes2 = model2.graph.node
+    nodes2_names = [node.name for node in nodes2]
+
+    # Extract ops which are in nodes1 but not in nodes2
+    nodes1_names_diff = [
+        node_name for node_name in nodes1_names if node_name not in nodes2_names
+    ]
+
+    # Extract ops which are in nodes2 but not in nodes1
+    nodes2_names_diff = [
+        node_name for node_name in nodes2_names if node_name not in nodes1_names
+    ]
+    # Assert that there are no important ops names in
+    # nodes1_names_diff or nodes2_names_diff
+    assert not [
+        x for x in nodes1_names_diff if x.split("_")[0] not in optional_nodes_model1
+    ]
+    assert not [
+        x for x in nodes2_names_diff if x.split("_")[0] not in optional_nodes_model2
+    ]
+
+    # Compare the structure of nodes which share names across m1 and m2
+    for node1 in nodes1:
+        if node1.name in set(nodes1_names).intersection(set(nodes2_names)):
+            for node2 in nodes2:
+                if node1.name == node2.name:
+                    _compare_onnx_nodes(node1, node2)
+
+
+def _compare_onnx_nodes(n1, n2):
+    # checking for consistent lengths seems like a sufficient test for now.
+    # due to internal structure, the naming of graph nodes
+    # may vary, even thought the semantics remain unchanged.
+    assert len(n1.input) == len(n2.input)
+    assert len(n1.output) == len(n2.output)
+    assert len(n1.op_type) == len(n2.op_type)
+    assert len(n1.attribute) == len(n2.attribute)
+
+
+@pytest.mark.parametrize(
+    "model_stub, recipe_present, task",
+    [
+        (
+            "zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned-conservative",  # noqa: E501
+            False,
+            "question-answering",
+        )
+    ],
+    scope="function",
+)
+class TestModelFromZoo:
+    @pytest.fixture()
+    def setup(self, model_stub, recipe_present, task):
+        # setup
+        model = Zoo.load_model_from_stub(model_stub)
+        model.download()
+
+        path_onnx = model.onnx_file.downloaded_path()
+        model_path = os.path.join(os.path.dirname(path_onnx), "pytorch")
+
+        yield path_onnx, model_path, recipe_present, task
```

Comment:
is there a finalize/teardown method we can use for this class instead of using this yield pattern?

Reply:
Why should we do that? Using `yield` is the recommended way to do setup/teardown in pytest (https://docs.pytest.org/en/latest/how-to/fixtures.html#teardown-cleanup-aka-fixture-finalization). Tbh I find it more concise and elegant than writing two, separate methods for setup/teardown, so why change it?

But to answer your question, yes, there is a possibility to do it as you suggest: https://stackoverflow.com/questions/26405380/how-do-i-correctly-setup-and-teardown-for-my-pytest-class-with-tests

Reply:
seems a bit less intuitive, but yes, more concise, good find